### ISSUE TITLE
Computation of conditional probability

### ISSUE THREAD

**Issue by SamarthGarg09** [JiaQiSJTU/FaithEval-FFLM#2]
As it can be seen in equation 5 in the paper, the computation of conditional probability is the difference between S2S and Pref, however in the code it is vice versa pref-s2s. What is the correct equation, can you clarify?

**Comment by JiaQiSJTU** [comment]
Sure.
In equation 5, we explained the intuition of using probability changes where the equation is p_s2s - p_pref. 
Then, in sec 2.2, we introduced a feasible design based on those probability changes where we take the logarithm of the probabilities before subtraction: log p_s2s - log p_s2s.
In the code (utils.py line 51-55), we calculated the loss of the probability in advance, so that the final calculation is prefix_loss - s2s_loss, which equals - (-log p_s2s) + (- log p_s2s) = log p_s2s - log p_s2s.

**Comment by JiaQiSJTU** [comment]
Sure.
In equation 5, we explained the intuition of using probability changes where the equation is p_s2s - p_pref.
Then, in sec 2.2, we introduced a feasible design based on those probability changes where we take the logarithm of the probabilities before subtraction: log p_s2s - log p_pref.
In the code (utils.py line 51-55), we calculated the loss of the probability in advance, so that the final calculation is prefix_loss - s2s_loss, which equals - (-log p_s2s) + (- log p_ref) = log p_s2s - log p_pref.
